pyo3-macros-backend/src/method.rs

```diff
@@ -513,15 +513,15 @@ impl<'a> FnSpec<'a> {
                         holders.pop().unwrap(); // does not actually use holder created by `self_arg`
 
                         quote! {{
-                            let __guard = _pyo3::impl_::coroutine::RefGuard::<#cls>::new(py.from_borrowed_ptr::<_pyo3::types::PyAny>(_slf))?;
+                            let __guard = _pyo3::impl_::coroutine::RefGuard::<#cls>::new(_pyo3::Bound::from_borrowed_ptr(py, _slf))?;
```

Comment:
Today I had the thought that we could also use `BoundRef` here, to defer the ref count bump until after the `downcast` check in `RefGuard::new`